src/corehost/cli/corehost_context_contract.h

```diff
@@ -22,7 +22,8 @@ enum class coreclr_delegate_type
     load_in_memory_assembly,
     winrt_activation,
     com_register,
-    com_unregister
+    com_unregister,
+    component_create_native_delegate
```

Comment:
I think this should have something like "load assembly" in the name - since the runtime helper loads the assembly and then gets a delegate from it. In a way the "loading assembly" part is more important as it lets one inject managed code into any process.
I also don't like the "delegate" part too much - delegates are a managed concept. I know that this is similar to preexisting `coreclr_get_delegate` but I still think it's confusing.

So maybe something like `load_assembly_and_get_function_pointer`...

Reply:
Yeah I prefer having the 'load assembly' as well. I like `load_assembly_and_get_function_pointer` (a bit on the long side, but I can't think of anything shorter without losing useful information).